frontend/hub/collections/hooks/useDeleteCollectionsFromRepository.tsx

```diff
@@ -1,53 +1,122 @@
+        ? t('Permanently delete collections versions from repository', {
+            count: collections.length,
+          })
+        : t('Permanently delete collections from repository', { count: collections.length });
+
+      const confirmText = version
+        ? t(
+            'Yes, I confirm that I want to delete these {{count}} collections versions from repository.',
+            {
+              count: collections.length,
+            }
+          )
+        : t('Yes, I confirm that I want to delete these {{count}} collections from repository.', {
+            count: collections.length,
+          });
+
+      const actionButtonText = version
+        ? t('Delete collections versions', { count: collections.length })
+        : t('Delete collections', { count: collections.length });
+
       bulkAction({
-        title: t('Permanently delete collections from repositories', { count: collections.length }),
-        confirmText: t('Yes, I confirm that I want to delete these {{count}} collections.', {
-          count: collections.length,
-        }),
-        actionButtonText: t('Delete collections', { count: collections.length }),
+        title,
+        confirmText,
+        actionButtonText,
         items: collections.sort((l, r) =>
           compareStrings(
-            l.collection_version?.name || '' + l.repository?.name,
-            r.collection_version?.name || '' + r.repository?.name
+            l.collection_version?.name || '' + l.repository?.name + l.collection_version?.version,
+            r.collection_version?.name || '' + r.repository?.name + r.collection_version?.version
           )
         ),
         keyFn: collectionKeyFn,
         isDanger: true,
         confirmationColumns,
         actionColumns,
         onComplete,
-        actionFn: (collection: CollectionVersionSearch) =>
-          deleteCollectionFromRepository(collection),
+        actionFn: (collection: CollectionVersionSearch) => {
+          return deleteCollectionFromRepository(collection, version).then(() => {
+            if (detail) {
+              return navigateAfterDelete(collection, version || false, navigate);
+            }
+          });
+        },
       });
     },
-    [actionColumns, bulkAction, confirmationColumns, onComplete, t]
+    [actionColumns, bulkAction, confirmationColumns, onComplete, t, version, navigate, detail]
   );
 }
 
-async function deleteCollectionFromRepository(collection: CollectionVersionSearch) {
-  return postRequest(
+async function deleteCollectionFromRepository(
+  collection: CollectionVersionSearch,
+  version?: boolean
+) {
+  let itemsToDelete: string[] = [];
+
+  if (!version) {
+    // load all associated collection versions
+    const results = await requestGet<HubItemsResponse<CollectionVersionSearch>>(
+      hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${
+        collection.collection_version?.name || ''
+      }&repository_name=${collection.repository?.name || ''}`
```

Comment:
this function probably shouldn't try to call any apis if `collection.repository` is null?

Reply:
I will add the condition for repository, but when repo is null, user should not see the collection in UI anyway.